HACKING.md

```diff
@@ -0,0 +1,83 @@
+# Tips and tricks for hacking on Buck2
+
+You might have been lead here by reading [CONTRIBUTING.md](/CONTRIBUTING.md). If
+not, please read that as well! That will give you the high level overview; this
+is the document is all about the needed elbow grease you'll have to apply.
+
+## Building the code
+
+Buck2 is written in Rust, **and currently requires a nightly toolchain**. You
+need **[rustup](https://rustup.rs)** installed to provision the necessary
+version of `rustc` and `cargo` as a result.
+
+You can either build `buck2` from a clone of the Buck2 repo (this will cause
+rustup/cargo to install the right nightly version of `rustc`):
+
+```sh
+git clone https://github.com/facebook/buck2.git
+cd buck2/
+cargo install --path=app/buck2
+```
+
+Or alternatively install it directly from GitHub:
+
+```sh
+rustup install nightly-2023-03-07
+cargo +nightly-2023-03-07 install --git https://github.com/facebook/buck2.git buck2
+```
+
+Build uses prebuilt `protoc` binary from
+[protoc-bin-vendored](https://crates.io/crates/protoc-bin-vendored) crate.
```

Comment:
Not sure if this is important enough to mention here, but you can also override this with an environment variable, which was important to get it working on my platform.

Reply:
I've also reworked this into a side note about alternative operating systems and the need for `protoc`. Could be more organized but I think this should help.